respa_admin/static_src/js/periods.js

```diff
@@ -517,20 +517,9 @@ function removePeriodEventHandlers(periodItem) {
 }
 
 function removePeriod(periodItem) {
+  // checks the DELETE input flag for this item.
   if (periodItem) {
-    periodItem.remove();
```

Comment:
We want to hide the Period form and just flag it for deletion by setting the `DELETE` hidden input to a non-empty value. Django will then process the form and handle the deletion normally, and the original order of existing forms is maintained.